src/dotnet/commands/dotnet-add-p2p/Program.cs

```diff
@@ -0,0 +1,232 @@
+        // I.e. it returns null for some valid projects
+        public static ProjectRootElement TryOpenProject(string filename)
+        {
+            try
+            {
+                return ProjectRootElement.Open(filename);
+            }
+            catch (Microsoft.Build.Exceptions.InvalidProjectFileException)
+            {
+                return null;
+            }
+        }
+
+        public static ProjectRootElement GetProjectFromFileOrThrow(string filename)
+        {
+            if (!File.Exists(filename))
+            {
+                throw new GracefulException($"Provided project `{filename}` does not exist.");
+            }
+
+            var project = TryOpenProject(filename);
+            if (project == null)
+            {
+                throw new GracefulException($"Invalid MSBuild project `{filename}`.");
+            }
+
+            return project;
+        }
+
+        public static ProjectRootElement GetProjectFromCurrentDirectoryOrThrow()
+        {
+            DirectoryInfo currDir = new DirectoryInfo(Directory.GetCurrentDirectory());
+            FileInfo[] files = currDir.GetFiles("*proj");
+            if (files.Length == 0)
+            {
+                throw new GracefulException("Could not find any MSBuild project in the current directory.");
+            }
+
+            if (files.Length > 1)
+            {
+                throw new GracefulException("Found more than one MSBuild project in the current directory. Please specify which one to use.");
+            }
+
+            FileInfo projectFile = files.First();
+
+            if (!projectFile.Exists)
+            {
+                throw new GracefulException("Could not find any project in the current directory.");
+            }
+
+            var ret = TryOpenProject(projectFile.FullName);
+            if (ret == null)
+            {
+                throw new GracefulException($"Found an MSBuild project `{projectFile.FullName}` in the current directory but it is invalid.");
+            }
+
+            return ret;
+        }
+
+        public static Func<T, bool> AndPred<T>(params Func<T, bool>[] preds)
+        {
+            return (el) => preds.All((pred) => pred == null || pred(el));
+        }
+
+        public static string GetFrameworkConditionString(string framework)
+        {
+            if (string.IsNullOrEmpty(framework))
+            {
+                return null;
+            }
+
+            return $" '$(TargetFramework)' == '{framework}' ";
+        }
+
+        public static Func<T, bool> FrameworkPred<T>(string framework) where T : ProjectElement
+        {
+            string conditionStr = GetFrameworkConditionString(framework);
+            if (conditionStr == null)
+            {
+                return (ig) => {
+                    var condChain = ig.ConditionChain();
+                    return condChain.Count == 0;
+                };
+            }
+
+            conditionStr = conditionStr.Trim();
```

Comment:
This function is being used in both generating the condition and comparing them. When you generate it you expect spaces around as that's what we do everywhere, when comparing, you can't expect user to do that. I think it is perfectly fine to assume that this will not add any spaces so I'll do it as it in fact may seem counterintuitive.